libdrgn/python/type.c

```diff
@@ -1607,16 +1833,27 @@ static DrgnType *Program_compound_type(Program *self, PyObject *args,
 	    (can_cache_members &&
 	     _PyDict_SetItemId(type_obj->attr_cache, &DrgnType_attr_members.id,
 			       cached_members ?
-			       cached_members : Py_None) == -1))
+			       cached_members : Py_None) == -1) ||
+	    (can_cache_templates &&
+	     _PyDict_SetItemId(type_obj->attr_cache, &DrgnType_attr_template_parameters.id,
```

Comment:
I'm not sure if I'm doing this right - can you take a closer look at this (and the function version as well). I have no idea how to test it.